relayer/processor/path_processor.go

```diff
@@ -109,6 +127,17 @@ type channelPair struct {
 	pathEnd2ChannelKey ChannelKey
 }
 
+// RelevantClientID returns the relevant client ID or panics
+func (pp *PathProcessor) RelevantClientID(chainID string) string {
+	if pp.pathEnd1.info.ChainID == chainID {
+		return pp.pathEnd1.info.ClientID
+	}
+	if pp.pathEnd2.info.ChainID == chainID {
+		return pp.pathEnd2.info.ClientID
+	}
+	panic(fmt.Sprintf("no relevant client ID for chain ID: %s", chainID))
```

Comment:
Conventionally, you would pass an error (with `fmt.Errorf` here) to panic. A `recover` call elsewhere would typically expect an error value, not a string.

Reply:
+1 I kind of wish `panic` required an error but I'm sure they had their reasons.

Reply:
done